crypto/test/keys.go

```diff
@@ -39,3 +41,16 @@ func GenerateECKey() *ecdsa.PrivateKey {
 	key, _ := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
 	return key
 }
+
+// StringNamingFunc can be used to give a key a simple string name
+func StringNamingFunc(name string) storage.KIDNamingFunc {
+	return func(key crypto.PublicKey) (string, error) {
+		return name, nil
+	}
+}
+
+func ErrorNamingFunc(err error) storage.KIDNamingFunc {
```

Comment:
replacing `KIDNamingFunc` with `func(key crypto.PublicKey) (string, error)` removes cyclical dependencies so `KIDNamingFunc` can be with the rest in `spi` (or `storage`)

Reply:
-60 lines changed 👍